### PR TITLE
feat: audit data extractor version 1.0

### PR DESCRIPTION
**Pull request by davinerd**
This MR points to version 1.0 of the script with lots of improvements.

Please refer to README for more information.

### CODE REVIEW

File: tools/dhis2-audit-data-extractor/COPYING [hunk 1]
```diff
@@ -0,0 +1,674 @@
+                    GNU GENERAL PUBLIC LICENSE
```
Comment (mediremi) [inline tools/dhis2-audit-data-extractor/COPYING:1]: Normally our software is released under the New BSD 3-Clause license (see [here](https://github.com/dhis2/dhis2-core/blob/master/LICENSE)), so maybe we could use that license instead?

File: tools/dhis2-audit-data-extractor/README.md [hunk 2]
```diff
@@ -11,14 +23,61 @@ $ pip install -r requirements.txt
...
+$ python extract_audit.py extract -f JSON -o results.json -e 10
+Data saved in results.json
+$ cat results.json | jq length
+10
+$ python extract_audit.py extract -f JSON -o results.json -e 10 -s 10
+Data saved in results.json
+$ cat results.json | jq length
+10
+10
+$
```
Comment (mediremi) [inline tools/dhis2-audit-data-extractor/README.md:80]: The documentation and examples you've written are super clear and easy to understand :+1:  :star2:

File: tools/dhis2-audit-data-extractor/extract_audit.py [hunk 3]
```diff
@@ -1,11 +1,21 @@
...
-from sqlalchemy import create_engine
+#from memory_profiler import profile
+#import pandas as pd
+#from sqlalchemy import create_engine
+import argparse
+from datetime import datetime
+import csv
 
 DHIS2_HOME = os.getenv("DHIS2_HOME", "/home/dhis")
 DHIS2_CONF_FILE =  "{0}/config/dhis.conf".format(DHIS2_HOME)
```
Comment (mediremi) [inline tools/dhis2-audit-data-extractor/extract_audit.py:21]: As a future improvement, it'd be nice if we check early on if `DHIS2_HOME` and `DHIS2_CONF_FILE` exist and if not output an error to the user.

File: tools/dhis2-audit-data-extractor/extract_audit.py [hunk 4]
```diff
@@ -109,12 +67,133 @@ def extract_data():
...
+                    "type": row[2],
+                    "datetime": row[3].strftime("%Y-%m-%d %H:%M:%S"),
+                    "createdby": row[4],
+                    "klass": row[5],
+                    "uid": row[6],
+                    "code": row[7],
+                    "attributes": row[8],
+                    "data": json.loads(gzip.decompress(row[9]).decode('utf-8')) if row[9] else None
+                }
+            audit_data.append(event)
```
Comment (mediremi) [inline tools/dhis2-audit-data-extractor/extract_audit.py:166]: I'd suggest extracting the event dictionary creation part into a separate function to allow the code to focus more on the 'what' than the 'how' of the output:

```suggestion
        def get_event(row):
            return {
                    "id": row[0],
                    "event": row[1],
                    "type": row[2],
                    "datetime": row[3].strftime("%Y-%m-%d %H:%M:%S"),
                    "createdby": row[4],
                    "klass": row[5],
                    "uid": row[6],
                    "code": row[7],
                    "attributes": row[8],
                    "data": json.loads(gzip.decompress(row[9]).decode('utf-8')) if row[9] else None
            }

        with open(filename, 'a') as fd:
            writer = csv.DictWriter(fd, fieldnames = ["id", "event", "type", "datetime", "createdby", "klass", "uid", "code", "attributes", "data"])
            if format == "CSV" and not os.path.exists(filename):
                writer.writeheader()

            for row in iter_row(cur):
                event = get_event(row)
                if format == "CSV":
                    writer.writerow(event)
                elif format == "JSON":
                    audit_data.append(event)

            if format == "JSON":
                json.dump(audit_data, fd)
        print("Data saved in {0}".format(filename))
    elif output_mode == "stdout":
        for row in iter_row(cur):
            event = get_event(row)
            if format == "CSV":
                line = """{0},{1},{2},{3},{4},{5},{6},{7},'{8}','{9}'""".format(
                    event.id, event.event, event.type, event.datetime, event.createdby, event.klass, event.uid, event.code, event.attributes, event.data
                )
            elif format == "JSON":
                line = event
            audit_data.append(line)
```

File: tools/dhis2-audit-data-extractor/COPYING (hunk 1 above)
Comment (davinerd) [reply tools/dhis2-audit-data-extractor/COPYING:1]: Yes we do, but in this repo there isn't an established license. Plus, I don't feel comfortable with the BSD clause for a software used as public good, so I'd prefer more open licenses like GPL3 or even [CC4r](https://constantvzw.org/wefts/cc4r.en.html).

So unless there is a strong need to align licenses, I'd keep GPL3.

File: tools/dhis2-audit-data-extractor/extract_audit.py (hunk 3 above)
Comment (davinerd) [reply tools/dhis2-audit-data-extractor/extract_audit.py:21]: Added in the TODO list, thanks!

File: tools/dhis2-audit-data-extractor/extract_audit.py (hunk 4 above)
Comment (davinerd) [reply tools/dhis2-audit-data-extractor/extract_audit.py]: This [23d05de](https://github.com/dhis2/dhis2-utils/pull/57/commits/23d05deff8e5cadc73c882db5eec6d4c611c2421) implements your suggestion.

**Review by mediremi — APPROVED** [review]

**Pull request merged**